### PR TITLE
[WIP] [Experiment] feat(1534): Add "sealed" key word to mark some public/protected methods not be overriden

### LINKED ISSUE #1534: Add "sealed" key word to mark some public/protected methods not be overriden
From https://github.com/Microsoft/TypeScript/issues/1524, I see that the default method are all "virtual", but sometimes I don't wanna some public method to be overidden. Would you invent a mechasim to force the users that they cannot override a public method (with the key word, something like "sealed")?

Thanks!

### PR DESCRIPTION
**Pull request by a-tarasyuk**
This is a draft/proposal PR for adding a `sealed` modifier to prevent property/method overriding in child classes.
Any feedback to make a clear proposal/behavior is appreciated.

1. `sealed` cannot be used with `private`
2. `sealed` cannot be used with abstract declarations
3. `sealed` cannot be used with constructor declaration
4. `public`/`protected` `methods`/`properties` marked `sealed` cannot be overridden in child classes
5. TODO

Fixes #1534

### CODE REVIEW

**Comment by jjangga0214** [comment]
You can utilize a ["draft PR"](https://github.blog/2019-02-14-introducing-draft-pull-requests/) feature of GitHub!

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -41357,12 +41378,15 @@ namespace ts {
...
                     return grammarErrorOnNode(lastOverride!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "override"); // TODO: GH#18217
                 }
-                else if (flags & ModifierFlags.Async) {
+                if (flags & ModifierFlags.Async) {
                     return grammarErrorOnNode(lastAsync!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "async");
                 }
-                else if (flags & ModifierFlags.Readonly) {
+                if (flags & ModifierFlags.Readonly) {
                     return grammarErrorOnNode(lastReadonly!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "readonly");
                 }
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:41386]: This seems like it deserves to move to its own PR

Comment (a-tarasyuk) [reply src/compiler/checker.ts]: @DanielRosenwasser I removed the `noise` changes
